Open Library Service

cquiro opened the pull request:
## Summary

- Add httparty to Gemfile and bundle install
- Create OpenLibraryService with book_info method using httparty
- Create BookInfoController with show method that uses OpenLibraryService#book_info
- Create route to book_info#show

## Trello Card

https://trello.com/c/3Goo2Qll/38-integraci%C3%B3n-api-externa

## Screenshot

<img width="1112" alt="screen shot 2018-10-03 at 9 23 37 am" src="https://user-images.githubusercontent.com/12885969/46410123-1dd7d400-c6ee-11e8-9ade-0522952c5417.png">

cquiro committed:
Add httparty to Gemfile and bundle install

cquiro committed:
Create OpenLibraryService with book_info method.

cquiro committed:
Create endpoint to get book info by isbn

- Create BookInfoController with show method that uses
OpenLibraryService#book_info

- Create route to book_info#show

cquiro committed:
Fix rubocop offenses in OpenLibraryService.

cquiro committed:
Update the README with books/info_by_isbn endpoint

facucompe commented on app/services/open_library_service.rb:
```diff
@@ -0,0 +1,31 @@
+class OpenLibraryService
+  include HTTParty
+  base_uri 'https://openlibrary.org'
```
Use an env var

facucompe commented on app/services/open_library_service.rb:
```diff
@@ -0,0 +1,31 @@
+  attr_reader :isbn
+
+  def send_request
+    self.class.get('/api/books', query: { bibkeys: "ISBN:#{isbn}", format: 'json', jscmd: 'data' })
```
Move this to a variable like `GET_BOOKS = '/api/books'`

facucompe commented on app/services/open_library_service.rb:
```diff
@@ -0,0 +1,31 @@
+
+  def book_info
+    info = send_request.parsed_response
+    formatted_response(info)
```
I think you should use `info.body` here

cquiro replied:
@facucompe I get 'nil' because I'm not getting a 'body' element from `send_request.parsed_response`. This is info (`info = send_request.parsed_response`):

<img width="1399" alt="screen shot 2018-10-04 at 10 05 31 am" src="https://user-images.githubusercontent.com/12885969/46475952-4c70b000-c7bd-11e8-930e-cda4733b1dd3.png">

facucompe commented on config/routes.rb:
```diff
@@ -7,7 +7,9 @@
   api_version(module: 'Api::V1', path: { value: 'api/v1' }, defaults: { format: :json })  do
-    resources :books, only: [:index, :show]
+    resources :books, only: [:index, :show] do
+      get 'info_by_isbn', to: 'book_info#show', on: :collection
```
I would use the book controller too for this endpoint, but it's ok like this

cquiro committed:
Use variables in OpenLibraryService

- Use credentials to keep the OpenLibraryService base_uri.

- Create GET_BOOK constant to keep the '/api/books' uri.

cquiro committed:
Fix rubocop offenses in OpenLibraryService

- Freeze content assigned to constant.

- Remove byebug line from the code.

lmasello reviewed (approved):
Just facu´s comments and then it's ready to go 👍
